capture_hdr_loop.py

```diff
@@ -1,68 +1,65 @@
+    settings = zivid.Settings(
+        brightness=settings_from_yaml["Brightness"],
+        exposure_time=datetime.timedelta(
+            microseconds=settings_from_yaml["ExposureTime"]
+        ),
+        gain=settings_from_yaml["Gain"],
+        iris=settings_from_yaml["Iris"],
+    )
+    filters = settings_from_yaml["Filters"]
+    settings.filters.contrast.enabled = filters["Contrast"]["Enabled"]
+    settings.filters.contrast.threshold = filters["Contrast"]["Threshold"]
+    settings.filters.gaussian.enabled = filters["Gaussian"]["Enabled"]
+    settings.filters.gaussian.sigma = filters["Gaussian"]["Sigma"]
+    settings.filters.outlier.enabled = filters["Outlier"]["Enabled"]
+    settings.filters.outlier.threshold = filters["Outlier"]["Threshold"]
+    settings.filters.reflection.enabled = filters["Reflection"]["Enabled"]
+    settings.filters.saturated.enabled = filters["Saturated"]["Enabled"]
+    return settings
+
+
 def _main():
     app = zivid.Application()
 
     print("Connecting to the camera")
     camera = app.connect_camera()
 
-    print("Configuring the camera settings")
-    settings_collection = [camera.settings for _ in range(3)]
-
-    settings_collection[0].brightness = 1.8
-    settings_collection[0].gain = 1
-    settings_collection[0].filters.gaussian.enabled = True
-    settings_collection[0].filters.gaussian.sigma = 1.5
-    settings_collection[0].filters.reflection.enabled = True
-
-    settings_collection[1].brightness = 1.8
-    settings_collection[1].gain = 1
-    settings_collection[1].filters.gaussian.enabled = True
-    settings_collection[1].filters.gaussian.sigma = 1.5
-    settings_collection[1].filters.reflection.enabled = True
-
-    settings_collection[2].brightness = 1.8
-    settings_collection[2].gain = 1
-    settings_collection[2].filters.gaussian.enabled = True
-    settings_collection[2].filters.gaussian.sigma = 1.5
-    settings_collection[2].filters.reflection.enabled = True
-
-    exposure_time_frame_1 = [
-        datetime.timedelta(microseconds=10000),
-        datetime.timedelta(microseconds=90000),
-        datetime.timedelta(microseconds=40000),
-    ]
-    exposure_time_frame_2 = [
-        datetime.timedelta(microseconds=40000),
-        datetime.timedelta(microseconds=10000),
-        datetime.timedelta(microseconds=90000),
-    ]
-    exposure_time_frame_3 = [
-        datetime.timedelta(microseconds=90000),
-        datetime.timedelta(microseconds=40000),
-        datetime.timedelta(microseconds=10000),
-    ]
-    iris_frames = [10, 20, 30]
-
-    for i in range(len(iris_frames)):
-        settings_collection[0].exposure_time = exposure_time_frame_1[i]
-        settings_collection[1].exposure_time = exposure_time_frame_2[i]
-        settings_collection[2].exposure_time = exposure_time_frame_3[i]
+    # Get current settings, to be modified before use
+    settings = [camera.settings for _ in range(3)]
```

Comment:
There is not need to do this. The Settings objects are overwritten in the loop below anyway. 
You may as well create an empty list within the `for hdr_index in range(3)` loop and append to it for each frame.

Reply:
So,
```
    # Initialize settings list
    settings = [list() for _ in range(3)]
```
?
Otherwise I'll have to create a variable to hold 3 (number of frames in the HDR).

A more radical change would be for the example to traverse the `settings/` folder and compose HDRs based on what it finds. I don't want to go there. In fact, @SatjaSivcev has proposed some other changes to this example (for C++, which we'll align with Python afterwards)